microsoft/testsuites/power/power.py

```diff
@@ -0,0 +1,680 @@
+                    "/etc/default/grub.d/50-cloudimg-settings.cfg",
+                    '^GRUB_HIDDEN_TIMEOUT=30',
+                    sudo=True
+                )
+                assert_that(_entry2).described_as(
+                    f"{_entry2} - Missing config update "
+                    "in 50-cloudimg-settings.cfg file"
+                ).is_true()
+
+                _entry3 = node.tools[Cat].read_with_filter(
+                    "/etc/default/grub.d/50-cloudimg-settings.cfg",
+                    '^GRUB_TIMEOUT=30',
+                    sudo=True
+                )
+                assert_that(_entry3).described_as(
+                    f"{_entry3} - Missing config update "
+                    "in 50-cloudimg-settings.cfg file"
+                ).is_true()
+
+                log.info(
+                    "Successfully updated 50-cloudimg-settings.cfg file"
+                    " with all three entries.\n"
+                    "Setup Hibernate Kernel Completed.")
+
+        #################################
+        #      SETUP HIBERNATE DONE     #
+        #################################
+
+        # Reboot VM to apply swap setup changes
+        log.info("Rebooting VM")
+        start_stop = node.features[StartStop]
+        start_stop.restart()
+        # Check VM status, confirm it's running
+        if not _is_vm_running(node, log, self.REBOOT_TIME_OUT):
+            raise LisaException("Can not identify VM status before hibernate.")
+
+        cmd_res = node.execute(
+            "hwclock --set --date='2033-07-01 12:00:00'",
+            shell=True,
+            sudo=True
+        )
+        before_time_stamp = _get_year(node)
+        log.debug(f"Got before time stamp: {before_time_stamp}")
+
+        node.execute_async(
+            'echo disk > /sys/power/state',
+            shell=True,
+            sudo=True
+        )
+
+        log.info(
+            "Sent hibernate command to the VM and continue checking its status"
+            " in every 15 seconds until 10 minutes timeout.")
+
+        # Verify the VM status
+        # Can not find if VM hibernation completion
+        # or not as soon as it disconnects the network. Assume it is in timeout.
+        timeout = 600
+        start_time = time.time()
+        vm_stopped = False
+        while (time.time() - start_time < timeout):
+            time.sleep(15)
+            if not _is_vm_running(node, log, self.REBOOT_TIME_OUT):
+                vm_stopped = True
+                break
+            else:
+                log.debug("VM is not stopped. Waiting for 15s.......")
+
+        if vm_stopped:
+            log.info(
+                "Verified VM status is stopped after hibernation command sent")
+        else:
+            raise LisaException(
+                "Failed to change VM status to be stopped"
+                " after hibernation command sent")
+
+        # Resume the VM
+        start_stop.start()
```

Comment:
How about just reboot? Why it needs to shutdown and then start?

Reply:
shutdown is the default mode(i.e. in /sys/power/disk file) and it works when other modes could not work for some reason.
any special reason that we should change to reboot mode?